cmake/CopyBuildProducts.cmake

```diff
@@ -0,0 +1,97 @@
+function(copy_build_products target)
+  set(valid_options "DEPENDENCY;DEPENDENCY_FILES;TARGET_LOCATION;CONDITION")
+  foreach(arg IN LISTS ARGN)
```

Comment:
Had to write my own argument parsing here because CMake ~~sucks~~ typically doesn't like you specifying args in this manner when you use `parse_arguments` or whatever command it is again. Specifying args in this manner is more convenient for us, even if the error handling is incredibly primitive (read: nonexistant) and will likely cause strange errors if used incorrectly.